src/app/vault/vault.component.ts

```diff
@@ -547,56 +534,73 @@ export class VaultComponent implements OnInit, OnDestroy {
     this.go();
   }
 
-  async clearGroupingFilters() {
-    this.searchBarService.setPlaceholderText(this.i18nService.t("searchVault"));
-    await this.ciphersComponent.reload();
-    this.clearFilters();
-    this.go();
-  }
-
-  async filterFavorites() {
-    this.searchBarService.setPlaceholderText(this.i18nService.t("searchFavorites"));
-    await this.ciphersComponent.reload((c) => c.favorite);
-    this.clearFilters();
-    this.favorites = true;
-    this.go();
-  }
-
-  async filterDeleted() {
-    this.searchBarService.setPlaceholderText(this.i18nService.t("searchTrash"));
-    this.ciphersComponent.deleted = true;
-    await this.ciphersComponent.reload(null, true);
-    this.clearFilters();
-    this.deleted = true;
+  async applyVaultFilter(vaultFilter: VaultFilter) {
+    this.searchBarService.setPlaceholderText(
+      this.i18nService.t(this.calculateSearchBarLocalizationString(vaultFilter))
+    );
+    this.activeFilter = vaultFilter;
+    await this.ciphersComponent.reload(this.buildFilter());
     this.go();
   }
 
-  async filterCipherType(type: CipherType) {
-    this.searchBarService.setPlaceholderText(this.i18nService.t("searchType"));
-    await this.ciphersComponent.reload((c) => c.type === type);
-    this.clearFilters();
-    this.type = type;
-    this.go();
-  }
+  private calculateSearchBarLocalizationString(vaultFilter: VaultFilter): string {
+    if (vaultFilter.status === "favorites") {
+      return "searchFavorites";
+    }
+    if (vaultFilter.status === "trash") {
+      return "searchTrash";
+    }
+    if (vaultFilter.cipherType != null) {
+      return "searchType";
+    }
+    if (vaultFilter.selectedFolderId != null && vaultFilter.selectedFolderId != "none") {
+      return "searchFolder";
+    }
+    if (vaultFilter.selectedCollectionId != null) {
+      return "searchCollection";
+    }
+    if (vaultFilter.selectedOrganizationId != null) {
+      return "searchOrganization";
+    }
+    if (vaultFilter.myVaultOnly) {
+      return "searchMyVault";
+    }
 
-  async filterFolder(folderId: string) {
-    folderId = folderId === "none" ? null : folderId;
-    this.searchBarService.setPlaceholderText(this.i18nService.t("searchFolder"));
-    await this.ciphersComponent.reload((c) => c.folderId === folderId);
-    this.clearFilters();
-    this.folderId = folderId == null ? "none" : folderId;
-    this.go();
+    return "searchVault";
   }
 
-  async filterCollection(collectionId: string) {
-    this.searchBarService.setPlaceholderText(this.i18nService.t("searchCollection"));
-    await this.ciphersComponent.reload(
-      (c) => c.collectionIds != null && c.collectionIds.indexOf(collectionId) > -1
-    );
-    this.clearFilters();
-    this.collectionId = collectionId;
-    this.updateCollectionProperties();
-    this.go();
+  private buildFilter(): (cipher: CipherView) => boolean {
```

Comment:
Should this be a public method on the filter class? It doesn't refer to any methods/properties on this class.